app/src/main/java/org/kiwix/kiwixmobile/nav/destination/reader/KiwixReaderFragment.kt

```diff
@@ -60,30 +60,54 @@ import org.kiwix.kiwixmobile.core.utils.TAG_KIWIX
 import org.kiwix.kiwixmobile.core.utils.UpdateUtils
 import org.kiwix.kiwixmobile.core.utils.files.FileUtils
 import org.kiwix.kiwixmobile.main.KiwixMainActivity
-import org.kiwix.kiwixmobile.navigate
 import java.io.File
 
 private const val HIDE_TAB_SWITCHER_DELAY: Long = 300
 
-class ReaderFragment : CoreReaderFragment() {
-  private val args: ReaderFragmentArgs by navArgs()
+class KiwixReaderFragment : CoreReaderFragment() {
 
   override fun inject(baseActivity: BaseActivity) {
     (baseActivity as KiwixMainActivity).cachedComponent.inject(this)
   }
 
   override fun onActivityCreated(savedInstanceState: Bundle?) {
     super.onActivityCreated(savedInstanceState)
-    manageExternalLaunchAndRestoringViewState(args.zimFileUri)
+
     val activity = activity as CoreMainActivity
     noOpenBookButton.setOnClickListener {
       activity.navigate(
-        ReaderFragmentDirections.actionNavigationReaderToNavigationLibrary()
+        KiwixReaderFragmentDirections.actionNavigationReaderToNavigationLibrary()
       )
     }
     activity.supportActionBar!!.setDisplayHomeAsUpEnabled(true)
     activity.setupDrawerToggle(toolbar)
     setFragmentContainerBottomMarginToSizeOfNavBar()
+    if (arguments != null) {
+      openPageInBookFromNavigationArguments()
+    }
+  }
+
+  private fun openPageInBookFromNavigationArguments() {
+    val args = KiwixReaderFragmentArgs.fromBundle(requireArguments())
+    if (args.pageUrl.isNotEmpty()) {
+      if (args.zimFileUri.isNotEmpty()) {
+        tryOpeningZimFile(args)
+      }
+      loadUrlWithCurrentWebview(args.pageUrl)
+      requireArguments().clear()
+    } else {
+      manageExternalLaunchAndRestoringViewState(args.zimFileUri)
```

Comment:
this now doesn't happen by default, arguments has to be not null. intentional?

Reply:
Thought it was nullable, perhaps not then.

Reply:
I don't know the Nav library well enough, I see that a default value has been supplied in the graph? Does that mean any navigation here has non null args?

Reply:
They don't use `Args.fromBundle` though. Seems like it is non-nullable.